utils/notificationhub.ts

```diff
@@ -0,0 +1,175 @@
+import { APNSPushType, INotificationTemplate } from "./notification";
+
+const hubName = getRequiredStringEnv("AZURE_NH_HUB_NAME");
+const endpointOrConnectionString = getRequiredStringEnv("AZURE_NH_ENDPOINT");
+
+const notificationHubService = azure.createNotificationHubService(
+  hubName,
+  endpointOrConnectionString
+);
+
+/**
+ * A template suitable for Apple's APNs.
+ *
+ * @see https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html
+ */
+const APNSTemplate: INotificationTemplate = {
+  body:
+    '{"aps": {"alert": {"title": "$(title)", "body": "$(message)"}}, "message_id": "$(message_id)"}'
+};
+
+/**
+ * Build a template suitable for Google's GCM.
+ *
+ * @see https://developers.google.com/cloud-messaging/concept-options
+ */
+const GCMTemplate: INotificationTemplate = {
+  body:
+    '{"data": {"title": "$(title)", "message": "$(message)", "message_id": "$(message_id)", "smallIcon": "ic_notification", "largeIcon": "ic_notification"}}'
+};
+
+// send the push notification only to the last
+// device that set the installationId
+// see https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-push-notification-registration-management#installations
+export const toNotificationTag = (fiscalCodeHash: FiscalCodeHash) =>
+  `$InstallationId:{${fiscalCodeHash}}`;
+
+const CreateOrUpdateInstallationOptions = t.interface({
+  installationId: t.string,
+  platform: t.string,
+  pushChannel: t.string,
+  tags: t.array(t.string),
+  templates: t.interface({
+    template: INotificationTemplate
+  })
+});
+
+type CreateOrUpdateInstallationOptions = t.TypeOf<
+  typeof CreateOrUpdateInstallationOptions
+>;
+
+const NotifyPayload = t.interface({
+  message: t.string,
+  message_id: t.string,
+  title: t.string
+});
+
+type NotifyPayload = t.TypeOf<typeof NotifyPayload>;
+// NH result
+const NHResultSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+
+type NHResultSuccess = t.TypeOf<typeof NHResultSuccess>;
+
+const successNH = () =>
+  NHResultSuccess.encode({
+    kind: "SUCCESS"
+  });
+
+export const notify = (
+  installationId: FiscalCodeHash,
+  payload: NotifyPayload
+) => {
+  return tryCatch(
+    () => {
+      return new Promise<NHResultSuccess>((resolve, reject) =>
+        notificationHubService.send(
+          toNotificationTag(installationId),
+          payload,
+          {
+            // Add required headers for APNS notification to iOS 13
+            // https://azure.microsoft.com/en-us/updates/azure-notification-hubs-updates-ios13/
+            headers: {
+              ["apns-push-type"]: APNSPushType.ALERT,
+              ["apns-priority"]: 10
+            }
+          },
+          (error, _) =>
+            error == null
+              ? resolve(successNH())
+              : reject(
+                  `Error while sending notification to NotificationHub| ${error.message}`
```

Comment:
```suggestion
                  `Error while sending notification to NotificationHub|${error.message}`
```

Reply:
@AleDore I think that the problem with the execution of the Unit Test could be this line in the Azure DevOps pipeline (it only installs the node module dependencies without making the build): 

https://github.com/pagopa/io-functions-app/blob/90e5a59bc1c19982b124d01a2b42d19a4ba26842/azure-pipelines.yml#L161

with the last changes the tests to be executed need the tsc generation? Try changing the above line with `make: build`...